core/src/main/scala/org/apache/spark/util/logging/FileAppender.scala

```diff
@@ -113,7 +123,18 @@ private[spark] class FileAppender(inputStream: InputStream, file: File, bufferSi
 private[spark] object FileAppender extends Logging {
 
   /** Create the right appender based on Spark configuration */
-  def apply(inputStream: InputStream, file: File, conf: SparkConf): FileAppender = {
```

Comment:
I suppose you could introduce a new optional argument at the end rather than overload. If it raises a MiMa error we can exclude it

Reply:
Done